src/hotspot/share/gc/shenandoah/shenandoahCollectionSet.hpp

```diff
@@ -47,6 +47,15 @@ class ShenandoahCollectionSet : public CHeapObj<mtGC> {
   size_t                _garbage;
   size_t                _used;
   size_t                _region_count;
+  size_t                _immediate_trash;
```

Comment:
Might consider having collection set itself manage these new members through `add_region` API - rather than having similar looking code in each of the heuristics when the collection set is populated.

Reply:
This is a good idea.  I'll work on this change.

Reply:
This change will be incorporated in next commit.